### PR TITLE
jrl-cmakemodules: init at 0-unstable-2024-04-12

### PR DESCRIPTION
**Pull request by nim65s**
## Description of changes

https://github.com/jrl-umi3218/jrl-cmakemodules is a collection of cmake helpers.
It is used as a git submodule in many projects, including eigenpy, hpp-fcl, pinocchio, crocoddyl, and others.

Packaging it will allow us to avoid the need to fetch the submodule in those projects, and prevent us from having many copies of of the same files, potentially in different versions, in nixpkgs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nim65s** [comment]
eigenpy just got a new release allowing us to use this feature: https://github.com/stack-of-tasks/eigenpy/releases/tag/v3.5.0

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 303339` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>jrl-cmakemodules</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.eigenpy</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.hpp-fcl</li>
    <li>python311Packages.hpp-fcl.dev</li>
    <li>python311Packages.hpp-fcl.doc</li>
    <li>python311Packages.pinocchio</li>
    <li>python312Packages.crocoddyl</li>
    <li>python312Packages.eigenpy</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.hpp-fcl</li>
    <li>python312Packages.hpp-fcl.dev</li>
    <li>python312Packages.hpp-fcl.doc</li>
    <li>python312Packages.pinocchio</li>
  </ul>
</details>

File: pkgs/by-name/jr/jrl-cmakemodules/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+}:
+
+stdenv.mkDerivation {
+  pname = "jrl-cmakemodules";
+  version = "unstable-2024-04-12";
```
Comment (wegank) [inline pkgs/by-name/jr/jrl-cmakemodules/package.nix:10]:
```suggestion
  version = "0-unstable-2024-04-12";
```

https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming

Comment (nim65s) [reply pkgs/by-name/jr/jrl-cmakemodules/package.nix]: Ok, thanks !

**Pull request merged**